internal/cephfs/fsjournal.go

```diff
@@ -179,3 +246,132 @@ func reserveVol(ctx context.Context, volOptions *volumeOptions, secret map[strin
 
 	return &vid, nil
 }
+
+// reserveSnap is a helper routine to request a UUID reservation for the CSI SnapName and,
+// to generate the snapshot identifier for the reserved UUID.
+func reserveSnap(ctx context.Context, volOptions *volumeOptions, parentSubVolName string, snap *cephfsSnapshot, cr *util.Credentials) (*snapshotIdentifier, error) {
```

Comment:
This function is almost a complete copy/paste from `reserveVol()`, please split up the functions in smaller helpers and re-use the code.

Reply:
I dont understand this comment. The `reserveSnap` connect to `snapJournal` and `reserveVol` connect to `volJournal.`.etc. The options passed for reservation are different too. Altoghther the code is 25-30 lines..

Reply:
I dont understand this comment. The `reserveSnap` connect to `snapJournal` and `reserveVol` connect to `volJournal.`.etc. The options passed for reservation are different too. Altoghther the code is 25-30 lines..

Reply:
This is mostly about the horrible `j.ReserveName()` function. Placing that in a helper that passes the common parameters will make it cleaner. Not a must, we can consider improving this later.